aea/aea_builder.py

```diff
@@ -604,6 +606,42 @@ def build(self, connection_ids: Optional[Collection[PublicId]] = None) -> AEA:
         self._load_and_add_skills(aea.context)
         return aea
 
+    # TODO: remove and replace with a clean approach (~noise based crypto module or similar)
+    def _update_identity(
+        self, identity: Identity, wallet: Wallet, connections: List[Connection]
+    ) -> Identity:
+        """
+        TEMPORARY fix to update identity with address from noise p2p connection. Only affects the noise p2p connection.
+        """
+        public_ids = []  # type: List[PublicId]
+        for connection in connections:
+            public_ids.append(connection.public_id)
+        if not PublicId("fetchai", "p2p_noise", "0.1.0") in public_ids:
+            return identity
+        if len(public_ids) == 1:
+            p2p_noise_connection = connections[0]
+            noise_addresses = {
+                p2p_noise_connection.noise_address_id: p2p_noise_connection.noise_address  # type: ignore
+            }
+            # update identity:
+            assert self._name is not None, "Name not set!"
+            if len(wallet.addresses) > 1:
+                identity = Identity(
+                    self._name,
+                    addresses={**wallet.addresses, **noise_addresses},
+                    default_address_key=p2p_noise_connection.noise_address_id,  # type: ignore
+                )
+            else:  # pragma: no cover
+                identity = Identity(self._name, address=p2p_noise_connection.noise_address)  # type: ignore
+            return identity
+        else:
+            logger.error(
+                "The p2p-noise connection can only be used as a single connection. "
```

Comment:
We cannot allow running p2p_noise with other connection atm as this would cause unexpected side effects due to the hack.